### PR TITLE
Adding a Boolean test transformer

### PR DESCRIPTION
**Pull request by TonioGela**
Solves #25. Thanks to @adrian-salajan for the implementation.

P.S. The implementation could have been more barebone, like 

```scala
  implicit def effectOfBooleanToPropF[F[_]](
      fu: F[Unit]
  )(implicit F: MonadError[F, Throwable]): PropF[F] =
    Suspend[F, Result[F]](
      fu.map(boolean(_))
        .handleError(t => Result[F](Prop.Exception(t), Nil, Set.empty, Set.empty))
    )
```

or similar, is this preferrable?

### CODE REVIEW

**Review by satorg — APPROVED** [review]
Looks good, thanks!

**Comment by TonioGela** [comment]
@satorg I swear I'll do my best to keep track of this PR, but if we don't merge it I fear I'll forget about it soon :D

**Comment by satorg** [comment]
@TonioGela , thank you for the ping.
Usually I don't merge if I'm the only reviewer – just to let someone else review it. But looks like this repository doesn't have that many active maintainers now to make it working.

Thank you for the contribution again – it can come handy in some cases indeed!

**Comment by TonioGela** [comment]
Thanks for merging it @satorg 

> But looks like this repository doesn't have that many active maintainers now to make it working.

That's precisely the reason I nudged this a bit. To avoid forgetting about it, not to be rude or hijack other maintainers' will. Thanks again.

**Pull request merged**
